src/components/node-list/node-list-row.test.js

```diff
@@ -85,6 +84,37 @@ describe('NodeListRow', () => {
         false
       );
     });
+
+    describe('focus mode', () => {
+      it('sets the visibility toggle to the checked mode when the row is selected for focusmode', () => {
+        const { props } = setupProps();
+        const wrapper = setup.mount(
+          <NodeListRow {...props} focusMode={node} type="modularPipeline" />
+        );
+
+        expect(
+          wrapper.find('.pipeline-row__toggle-icon--focus-checked').exists()
+        ).toBe(true);
+      });
+
+      it('sets the label to match the selected parent when the row belongs to a modular pipeline selected by focus mode', () => {
+        const selectedItem = { id: 'abc' };
+        const { props } = setupProps();
+        const wrapper = setup.mount(
+          <NodeListRow
+            {...props}
+            focusMode={selectedItem}
+            parentDisabled={true}
+            parentPipeline={'abc'}
+            disabled={true}
```

Comment:
Should we test a couple more combo of this condition? It's a complex condition in the code.

Reply:
Do you think it's worth integration testing at the `node-list-tree.js` level as well?

Reply:
Thanks for raising this - indeed that's a great point! 

In terms of testing on the `node-list-tree`, it's a great point that you raised this - I had looked into setting tests for that level as well, however there isn't anything to test on that level given that we are only passing in the `focusMode` data through that level to `node-list-row`, hence I didn't add any tests to that. 

To elaborate on the further tests needed to set up, there are 2 few key things in this PR that we need to set up tests for to ensure its functionality:

1. ensuring all actions, reducers and selectors relating to the selection of an individual modular pipeline works - those tests have all been previously set up when we set up the modular pipeline as tags concept. 
2. the behaviour of the individual row on the tree - the main addition of a condition where the row is `disabled` while it needing to appear in a 'non-disabled' way when it is part of a selected modularpipeline under focus mode - those are the tests set up here. 

Given that most of the tests have been set up previously, hence this is why the tests set up here does seem quite minimal 😁 However, I can  perhaps look into adding 1-2 more cases for node-list-row to cross test the non-selected focus mode scenerio as a cross reference to the current tests. 

Heads up: in my next PR I will set up a lot more selectors and actions for focusmode to include the new graph nodes selection for the related dataset nodes, which by then I do have a bunch of new updated tests coming up in the next PR once we add in a series of new actions to call the related database notes for the flowchart 😁 By then you would be able to see how the entire feature would be tested end to end.

Reply:
@limdauto I've added a bunch of tests to illustrate all the possible scenerios for node-list-row under this new setup in my latest commit